src/qt/walletframe.cpp

```diff
@@ -212,7 +212,7 @@ void WalletFrame::gotoLoadPSBT(bool from_clipboard)
             return;
         }
         std::ifstream in{filename.toLocal8Bit().data(), std::ios::binary};
-        data.assign(std::istream_iterator<unsigned char>{in}, {});
+        data.assign(std::istreambuf_iterator<char>{in}, {});
```

Comment:
This change reverts https://github.com/bitcoin/bitcoin/commit/a65931e3ce66d87b8f83d67ecdbb46f137e6a670 partially, in particular:
- `std::istream_iterator` --> `std::istreambuf_iterato`
- `unsigned char` --> `char`

But the local `std::vector data` is still parameterized by `unsigned char`: https://github.com/bitcoin-core/gui/blob/bb5ea1d9a954b7b9f443ee8fbbb04549cd0b08a7/src/qt/walletframe.cpp#L195 

Is it on purpose?

Reply:
Yes, that was intentional. Using `unsigned char` does not appear to work (fails to compile). But I don't think that data type makes a difference here.

Reply:
Yeah, `std::ifstream` is `std::basic_ifstream<char>`, and passing that to `std::istreambuf_iterator` can only be done if it is templated `<char>` as well. C++ (17?) derives this automatically, so you can also write:


```cpp
data.assign(std::istreambuf_iterator{in}, {});